### DISCUSSION TITLE
Using `prismaField` inside of `simpleObject`

### DISCUSSION THREAD

**Discussion by devunt** [hayes/pothos#574] [Q&A]
First of all, I'm really enjoying using pothos. Thank you for such a great library!

What is a right way to include `prismaField` inside of `simpleObject`?
I'm using `simpleObject` to pass prisma model and additional information to client in a single request.
Currently I'm using following code.

```js
const User = builder.prismaObject('User', {
  ...
});

export const UserWithAccessToken = builder.simpleObject('UserWithAccessToken', {
  fields: (t) => ({
    user: t.field({ type: User }),
    accessToken: t.string(),
  }),
});

builder.mutationFields((t) => ({
  login: t.field({
    type: UserWithAccessToken,
    resolve: async (...) => {
      const user = await db.user.findUniqueOrThrow(...);
      const accessToken = createAccessToken(user.id);
      return { user, accessToken };
    }.
  }),
});
```

It works well as expected, but as it is not using `prismaField` for declaring mutation, it seems it does not having benefit from prisma plugin's pre-load functionality and falls back to `findUnique` method which will issue a another query. Is there a way to somehow spread `query` into nested simple object and avoids issuing another query for resolving nested prisma object?

**Comment by hayes** [comment]
Short answer is that the simpleObject plugin is designed for simple objects that don't have fields that require resolvers.  The whole point is to NOT have things like prismaFields.  That being said, I can understand why this would be useful.  See [this comment](https://github.com/hayes/pothos/issues/512#issuecomment-1210150045) for an example of how to do this
